product-service/src/types/products.type.ts

```diff
@@ -7,6 +9,27 @@ export interface IProduct {
   image: string;
 }
 
+export interface ISeparateProduct {
+  id: string;
+  title: string;
+  description: string;
+  price: number;
+  image: string;
+}
+
+export const CreateProductSchema = Joi.object({
+  title: Joi.string().min(1).required(),
+  description: Joi.string().min(1).required(),
+  price: Joi.number().required(),
```

Comment:
Probably it would make sense to add minimal value restriction for price and count.

Reply:
You've got a point, added those changes just now, thanks